### PR TITLE
SLS-479 Fix macos build

### PR DESCRIPTION
**Pull request by quchenhao**
Summarize the reason behind this change (this might be the problem you're solving, or the context around the request) and the solution you have chosen.

### CODE REVIEW

File: src/block_disagg/block_disagg_read.c [hunk 1]
```diff
@@ -174,20 +174,21 @@ __block_disagg_read_multiple(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_di
...
+                      "page %" PRIu64 " ckpt %" PRIu64 ", magic %" PRIu8
                       ": doesn't match expected magic of %" PRIu8,
                       block_disagg->name, size, page_id, checkpoint_id, swap.magic, expected_magic);
                     goto corrupt;
                 }
-
-                if (swap.compatible_version > WT_BLOCK_DISAGG_COMPATIBLE_VERSION) {
+                /* TODO: workaround MACOS build failure when passing macro to a string format. */
+                compatibile_version = WT_BLOCK_DISAGG_COMPATIBLE_VERSION;
+                if (swap.compatible_version > compatibile_version) {
```
Comment (ddanderson) [inline src/block_disagg/block_disagg_read.c:184]: It might be nicer to
```
#define WT_BLOCK_DISAGG_COMPATIBLE_VERSION ((uint8_t)0x1u)
```
if that works.

**Review by ddanderson — APPROVED** [review]
I left one suggestion to consider, your choice.  I approve either way.

**Review by korteland — APPROVED** [review]
LGTM, I like Don's suggestion.

File: src/block_disagg/block_disagg_read.c (hunk 1 above)
Comment (quchenhao) [reply src/block_disagg/block_disagg_read.c:184]: I tired and it doesn't work on linux

Comment (quchenhao) [reply src/block_disagg/block_disagg_read.c]: done

**Pull request merged**